modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/metadata/MetadataOperationTestCase.java

```diff
@@ -126,16 +126,16 @@ public void getMultilevelKeys() throws Exception {
 
   @Test
   public void injectComposedMetadataKeyIdInstanceInMetadataResolver() throws Exception {
-    componentId = new ProcessorId(SIMPLE_MULTILEVEL_KEY_RESOLVER, FIRST_PROCESSOR_INDEX);
+    location = Location.builder().globalName(SIMPLE_MULTILEVEL_KEY_RESOLVER).addIndexPart(0).build();
     MetadataKey key = newKey(AMERICA, CONTINENT).withChild(newKey(USA, COUNTRY).withChild(newKey(SAN_FRANCISCO, CITY))).build();
     final MetadataResult<ComponentMetadataDescriptor<OperationModel>> metadataResult =
-        metadataService.getOperationMetadata(componentId, key);
+        metadataService.getOperationMetadata(location, key);
     assertSuccessResult(metadataResult);
   }
 
   @Test
   public void dynamicOperationMetadata() throws Exception {
-    componentId = new ProcessorId(CONTENT_AND_OUTPUT_METADATA_WITH_KEY_ID, FIRST_PROCESSOR_INDEX);
+    location = Location.builder().globalName(CONTENT_AND_OUTPUT_METADATA_WITH_KEY_ID).addIndexPart(0).build();
```

Comment:
Are locations created somwwhere else other than core code? If not, builder and implemetation should be here rather than in api